lib/matplotlib/colors.py

```diff
@@ -943,6 +943,26 @@ def _set_extremes(self):
             self._lut[self._i_over] = self._lut[self.N - 1]
         self._lut[self._i_bad] = self._rgba_bad
 
+    def with_alpha(self, alpha):
+        """
+        Return a copy of the colormap with a new uniform transparency.
+
+        Parameters
+        ----------
+        alpha : float
+             The alpha blending value, between 0 (transparent) and 1 (opaque).
+        """
+        if not isinstance(alpha, Real):
+            raise TypeError(
+                f"'alpha' must be numeric or None, not {type(alpha)}")
```

Comment:
This fits in a line, doesn't it?
```suggestion
            raise TypeError(f"'alpha' must be numeric or None, not {type(alpha)}")
```